directed.go

```diff
@@ -37,6 +37,77 @@ func (d *directed[K, T]) AddVertex(value T, options ...func(*VertexProperties))
 	return d.store.AddVertex(hash, value, properties)
 }
 
+func (d *directed[K, T]) UpdateVertex(existingHash K, value T, options ...func(*VertexProperties)) error {
+	_, _, err := d.store.Vertex(existingHash)
+	if err != nil {
+		return err
+	}
+
+	newHash := d.hash(value)
+	if existingHash != newHash {
+		_, _, err = d.store.Vertex(newHash)
+		if !errors.Is(err, ErrVertexNotFound) {
+			return ErrVertexAlreadyExists
+		}
+	}
+
+	adjacencyMap, err := d.AdjacencyMap()
+	if err != nil {
+		return err
+	}
+
+	existingEdges := adjacencyMap[existingHash]
+
+	for _, edge := range existingEdges {
+		err = d.RemoveEdge(edge.Source, edge.Target)
```

Comment:
For anyone who comes across this later - incoming edges (the ones in PredecessorMap()) also need to be deleted for RemoveVertex to work. Afaik the adjacencyMap only has outgoing ones